Crash Report 3.0.3: IllegalArgmentException FileUtils.contentQuery

**Describe the bug**
```
Caused by: java.lang.IllegalArgumentException: 

 
  at android.database.DatabaseUtils.readExceptionFromParcel (DatabaseUtils.java:165)

 
  at android.database.DatabaseUtils.readExceptionFromParcel (DatabaseUtils.java:135)

 
  at android.content.ContentProviderProxy.query (ContentProviderProxy.java:418)

 
  at android.content.ContentResolver.query (ContentResolver.java:816)

 
  at android.content.ContentResolver.query (ContentResolver.java:766)

 
  at android.content.ContentResolver.query (ContentResolver.java:717)

 
  at org.kiwix.kiwixmobile.utils.files.FileUtils.contentQuery (FileUtils.java:189)

 
  at org.kiwix.kiwixmobile.utils.files.FileUtils.getLocalFilePathByUri (FileUtils.java:165)

 
  at org.kiwix.kiwixmobile.main.MainActivity.manageExternalLaunchAndRestoringViewState (MainActivity.java:1891)

 
  at org.kiwix.kiwixmobile.main.MainActivity.onCreate (MainActivity.java:383)
```

**Expected behavior**
No crash

**Environment**
- Version of Kiwix Android : 3.0.3
- Device : Huawei P20 (HWEML)
- OS version : Android 9

--- a/app/src/main/java/org/kiwix/kiwixmobile/utils/files/FileUtils.kt
+++ b/app/src/main/java/org/kiwix/kiwixmobile/utils/files/FileUtils.kt
@@ -148,12 +148,12 @@ object FileUtils {
   }
 
   @JvmStatic fun getLocalFilePathByUri(
-    ctx: Context,
+    context: Context,
     uri: Uri
   ): String? {
 
     if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT &&
-      DocumentsContract.isDocumentUri(ctx, uri)
+      DocumentsContract.isDocumentUri(context, uri)
     ) {
       if ("com.android.externalstorage.documents" == uri.authority) {
         val documentId = DocumentsContract.getDocumentId(uri)
@@ -164,12 +164,12 @@ object FileUtils {
         }
       } else if ("com.android.providers.downloads.documents" == uri.authority)
         return try {
-          documentProviderContentQuery(ctx, uri)
+          documentProviderContentQuery(context, uri)
         } catch (ignore: IllegalArgumentException) {
           null
         }
     } else if ("content".equals(uri.scheme!!, ignoreCase = true)) {
-      return contentQuery(ctx, uri)
+      return contentQuery(context, uri)
     } else if ("file".equals(uri.scheme!!, ignoreCase = true)) {
       return uri.path
     }
@@ -177,9 +177,9 @@ object FileUtils {
   }
 
   @TargetApi(VERSION_CODES.KITKAT)
-  private fun documentProviderContentQuery(ctx: Context, uri: Uri) =
+  private fun documentProviderContentQuery(context: Context, uri: Uri) =
     contentQuery(
-      ctx,
+      context,
       ContentUris.withAppendedId(
         Uri.parse("content://downloads/public_downloads"),
         try {

--- a/app/src/main/java/org/kiwix/kiwixmobile/utils/files/FileUtils.kt
+++ b/app/src/main/java/org/kiwix/kiwixmobile/utils/files/FileUtils.kt
@@ -17,10 +17,12 @@
  */
 package org.kiwix.kiwixmobile.utils.files
 
+import android.annotation.TargetApi
 import android.content.ContentUris
 import android.content.Context
 import android.net.Uri
 import android.os.Build
+import android.os.Build.VERSION_CODES
 import android.os.Environment
 import android.provider.DocumentsContract
 import android.util.Log
@@ -160,19 +162,12 @@ object FileUtils {
         if (documentId[0] == "primary") {
           return "${Environment.getExternalStorageDirectory()}/${documentId[1]}"
         }
-      } else if ("com.android.providers.downloads.documents" == uri.authority
-      )
-        return contentQuery(
-          ctx,
-          ContentUris.withAppendedId(
-            Uri.parse("content://downloads/public_downloads"),
-            try {
-              DocumentsContract.getDocumentId(uri).toLong()
-            } catch (ignore: NumberFormatException) {
-              0L
-            }
-          )
-        )
+      } else if ("com.android.providers.downloads.documents" == uri.authority)
+        return try {
+          documentProviderContentQuery(ctx, uri)
+        } catch (ignore: IllegalArgumentException) {
+          null
+        }
     } else if ("content".equals(uri.scheme!!, ignoreCase = true)) {
       return contentQuery(ctx, uri)
     } else if ("file".equals(uri.scheme!!, ignoreCase = true)) {
@@ -180,6 +175,20 @@ object FileUtils {
     }
     return null
   }
+
+  @TargetApi(VERSION_CODES.KITKAT)
+  private fun documentProviderContentQuery(ctx: Context, uri: Uri) =
+    contentQuery(
+      ctx,
+      ContentUris.withAppendedId(
+        Uri.parse("content://downloads/public_downloads"),
+        try {
+          DocumentsContract.getDocumentId(uri).toLong()
+        } catch (ignore: NumberFormatException) {
+          0L
+        }
+      )
+    )
 
   private fun contentQuery(
     context: Context,

